ProjectLighthouse/Controllers/SlotsController.cs

```diff
@@ -45,18 +45,27 @@ public async Task<IActionResult> SlotsBy([FromQuery] string u)
         [HttpGet("s/user/{id:int}")]
         public async Task<IActionResult> SUser(int id)
         {
+            User? user = await this.database.UserFromRequest(this.Request);
+            if (user == null) return this.StatusCode(403, "");
+
             Token? token = await this.database.TokenFromRequest(this.Request);
             if (token == null) return this.BadRequest();
 
             GameVersion gameVersion = token.GameVersion;
 
-            Slot slot = await this.database.Slots.Where(s => s.GameVersion <= gameVersion)
+            Slot? slot = await this.database.Slots.Where(s => s.GameVersion <= gameVersion)
                 .Include(s => s.Creator)
                 .Include(s => s.Location)
                 .FirstOrDefaultAsync(s => s.SlotId == id);
 
             if (slot == null) return this.NotFound();
 
+            RatedLevel? ratedLevel = await this.database.RatedLevels.FirstOrDefaultAsync(r => r.SlotId == id && r.UserId == user.UserId);
+            if (ratedLevel != null) {
+                slot.YourRating = ratedLevel.RatingLBP1;
+                slot.YourDPadRating = ratedLevel.Rating;
+            }
+            
             return this.Ok(slot.Serialize());
```

Comment:
not entirely sure setting it on the slot directly is a good idea

maybe pass it in to `Slot.Serialize()` or make it a parameter similar to `Slot.Hearts`?

Reply:
Fixed in https://github.com/LBPUnion/project-lighthouse/pull/39/commits/f157647f689ae687bbb52a6b5b4e922e0d0083ee